bindings/go/constructor/spec/v1/resources/schema-2020-12.json

```diff
@@ -120,44 +120,88 @@
       ]
     },
     "sourceDefinition": {
-      "type": "object",
-      "description": "Definition of a source artifact used to generate resources",
-      "required": [
-        "name",
-        "version",
-        "type",
-        "access"
-      ],
-      "properties": {
-        "name": {
-          "description": "Name of the source",
-          "type": "string",
-          "$ref": "#/$defs/identityAttributeKey"
-        },
-        "extraIdentity": {
-          "description": "Additional identity attributes for the source",
-          "$ref": "#/$defs/identityAttribute"
-        },
-        "version": {
-          "description": "Version of the source",
-          "$ref": "#/$defs/relaxedSemver"
-        },
-        "type": {
-          "description": "Type of the source",
-          "$ref": "#/$defs/ocmType"
-        },
-        "labels": {
-          "description": "Labels associated with the source",
-          "type": "array",
-          "items": {
-            "$ref": "#/$defs/label"
+      "oneOf": [
+        {
+          "type": "object",
+          "description": "a source that has a predefined access specification",
+          "required": [
+            "name",
+            "version",
+            "type",
+            "access"
```

Comment:
Is this true? Isn't it either access or input?

Reply:
Wait, you can use anyOf in the required field though right? Something like 
```json
  "anyOf": [
    { "required": ["input"] },
    { "required": ["access"] }
  ]
```

No?